packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -150,6 +150,55 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
 /// ** See code in examples/api/lib/widgets/gesture_detector/gesture_detector.1.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// Why isn't my parent [GestureDetector] onTap method called?
+///
+/// Given a parent [GestureDetector] which defines an onTap callback and a child
+/// [GestureDetector] that also defines an onTap callback. When the inner
+/// [GestureDetector] is tapped, both [GestureDetector]s send a [GestureRecognizer]
+/// into the gesture arena because the pointer coordinates are in both
+/// [GestureDetector]s bounds. The child [GestureDetector] wins because it was
+/// the first detector to enter the arena and it got first dips because it is in front.
+/// For more information on gesture desambiguation see:
+/// [Gesture disambiguation](https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation)
+///
+/// Setting [GestureDetector.behavior] to [HitTestBehavior.opaque]
+/// or [HitTestBehavior.translucent] has no impact on parent-child relationships
+/// like in this case. To block hits on children use an [IgnorePointer] or [AbsorbPointer].
+///
+/// Some callbacks are called during the gesture arena resolution
+/// (for instance onTapDown). If the parent [GestureDetector] defines such callbacks,
+/// they will be called. Some callbacks are conditionnaly called, for instance
+/// [TapGestureRecognizer.onTapDown] triggers once a short timeout has elapsed.
+///
+/// {@tool dartpad}
+/// This example uses a [GestureDetector] that wraps a green [Container] and a second
+/// [GestureDetector] that wraps a yellow [Container]. The second [GestureDetector] is
+/// a child of the green [Container].
+/// Both [GestureDetector]s define an onTap callback. When the callback is called it
+/// adds a red border to the corresponding [Container].
+///
+/// When the green [Container] is tapped, it's parent [GestureDetector] enters
+/// the gesture arena. It wins because there is no competing [GestureDetector] and
+/// the green [Container] shows a red border.
+/// When the yellow [Container] is tapped, it's parent [GestureDetector] enters
+/// the gesture arena. The [GestureDetector] that wraps the green [Container] also
+/// enters the gesture arena (the pointer events coordinates are inside both
+/// [GestureDetector]s bounds). The [GestureDetector] that wraps the yellow [Container]
+/// wins because it was the first detector to enter the arena and it got first dips
+/// because it is in front.
+///
+/// This example sets [debugPrintGestureArenaDiagnostics] to true.
+/// This flag prints useful information about gesture arenas.
+///
+/// Changing the [GestureDetector.behavior] property to [HitTestBehavior.translucent]
+/// or [HitTestBehavior.opaque] has no impact: both [GestureDetector]s enter the
+/// gesture arena, only one wins.
```

Comment:
I think this could be discussed more above and benefit from a separate example, reading it now, I feel like I understand the first case, but now I am wondering why this is.

Reply:
@Piinks 
Yes, more information is needed here, like we say in french : ‘ça tombe comme un cheveux sur la soupe’ (‘it falls like a hair on the soup’) :-) As stated in the PR description, my main goal is to push a PR that will explain HitTestBehavior. 
Explaining HitTestBehavior clearly is not an easy task, so this first PR is meant to be a starting point (for me to get a better knowledge on Flutter documentation expectations, to find how much details should be shown on the example, etc). When the main PR (the one about HitTestBehavior) will be there, my plan is to update this paragraph with a link to the detailed explanations with diagrams and examples.
Do you think it’s an acceptable strategy ?